[2025-06-11T11:21:41.056+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-06-11T11:21:41.080+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: covid_data_pipeline.fetch_and_store_covid_data manual__2025-06-11T11:21:39.528630+00:00 [queued]>
[2025-06-11T11:21:41.089+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: covid_data_pipeline.fetch_and_store_covid_data manual__2025-06-11T11:21:39.528630+00:00 [queued]>
[2025-06-11T11:21:41.089+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 3
[2025-06-11T11:21:41.105+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): fetch_and_store_covid_data> on 2025-06-11 11:21:39.528630+00:00
[2025-06-11T11:21:41.114+0000] {warnings.py:110} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61: DeprecationWarning: This process (pid=1748) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2025-06-11T11:21:41.116+0000] {standard_task_runner.py:63} INFO - Started process 1749 to run task
[2025-06-11T11:21:41.117+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'covid_data_pipeline', 'fetch_and_store_covid_data', 'manual__2025-06-11T11:21:39.528630+00:00', '--job-id', '54', '--raw', '--subdir', 'DAGS_FOLDER/ covid_pipeline.py', '--cfg-path', '/tmp/tmpk3bzln_c']
[2025-06-11T11:21:41.119+0000] {standard_task_runner.py:91} INFO - Job 54: Subtask fetch_and_store_covid_data
[2025-06-11T11:21:41.169+0000] {task_command.py:426} INFO - Running <TaskInstance: covid_data_pipeline.fetch_and_store_covid_data manual__2025-06-11T11:21:39.528630+00:00 [running]> on host da27caf1d70a
[2025-06-11T11:21:41.276+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='covid_data_pipeline' AIRFLOW_CTX_TASK_ID='fetch_and_store_covid_data' AIRFLOW_CTX_EXECUTION_DATE='2025-06-11T11:21:39.528630+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-06-11T11:21:39.528630+00:00'
[2025-06-11T11:21:41.277+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-06-11T11:21:41.586+0000] {fetch_and_store.py:19} INFO - Fetched data: {'cases': 704753890, 'deaths': 7010681, 'recovered': 675619811, 'timestamp': datetime.datetime(2025, 6, 11, 11, 21, 41, 586125)}
[2025-06-11T11:21:41.591+0000] {fetch_and_store.py:53} INFO - ⏭️ No new data. Latest record matches current data. Skipping insert.
[2025-06-11T11:21:41.594+0000] {python.py:237} INFO - Done. Returned value was: None
[2025-06-11T11:21:41.596+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-06-11T11:21:41.612+0000] {taskinstance.py:1206} INFO - Marking task as SUCCESS. dag_id=covid_data_pipeline, task_id=fetch_and_store_covid_data, run_id=manual__2025-06-11T11:21:39.528630+00:00, execution_date=20250611T112139, start_date=20250611T112141, end_date=20250611T112141
[2025-06-11T11:21:41.690+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 0
[2025-06-11T11:21:41.714+0000] {taskinstance.py:3498} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-06-11T11:21:41.716+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
